### ISSUE TITLE
Please update address of contracts UiPoolDataProviderV3,UiIncentiveDataProviderV3

### ISSUE THREAD

**Issue by timqi** [bgd-labs/aave-address-book#633]
UiPoolDataProviderV3 [0xE92cd6164CE7DC68e740765BC1f2a091B6CBc3e4](https://arbiscan.io/address/0xE92cd6164CE7DC68e740765BC1f2a091B6CBc3e4#code)
UiIncentiveDataProviderV3 [0x9abADECD08572e0eA5aF4d47A9C7984a5AA503dC](https://arbiscan.io/address/0x9abADECD08572e0eA5aF4d47A9C7984a5AA503dC#code)

Addresses seem not updated according to official document from [https://docs.aave.com/developers/deployed-contracts/v3-mainnet/arbitrum](https://docs.aave.com/developers/deployed-contracts/v3-mainnet/arbitrum)

**Comment by sakulstra** [comment]
@timqi , @HugaidaS it's updated in 4.1, but would you mind telling us in which context you are using these providers?

We're considering the removal in one of the following releases as these providers are designed by avara for the avara ui and are quite suboptimal for most 3th party usecases we saw in the wild.
Would be interesting to understand "how"/in which context they are used so we can prepare proper migration docs.

**Comment by timqi** [comment]
I'm monitoring COIN positions put in aave in realtime

**Comment by HugaidaS** [comment]
We are integrating Aave markets in our app, so such info as apy, aToken and underlying asset is required. In order to get it, aave sdk says to use these providers to get the values we need.

**Comment by sakulstra** [comment]
Is the code using the providers open source somewhere so we could check it out?
Also @HugaidaS, with `aave-sdk` you mean this https://github.com/aave/aave-utilities?

---

Just to be clear, there is no "problem" per se with these ui libraries provided by avara.

It's just that they are quite suboptimal for almost every usecase:
- they fetch an extreme amount of data that most apps (from what we checked don't need)
- they don't distinguish between immutable and variable values, so you e.g. always fetch token decimals & symbols although they never will change - while usually used via external calls, the gas here doesn't matter, **but** we benchmarked these things a bit and even for off-chain calls these methods are extremely slow due to the amount of fetched data
